internal/cortex/querier/queryrange/instrumentation.go

```diff
@@ -65,3 +94,30 @@ func (c *NoopCollector) Before(ctx context.Context, method string, start time.Ti
 
 // After implements instrument.Collector.
 func (c *NoopCollector) After(ctx context.Context, method, statusCode string, start time.Time) {}
+
+// DurationHistogramCollector collects the duration of a request
+type DurationHistogramCollector struct {
+	metric *prometheus.HistogramVec
+}
+
+func (c *DurationHistogramCollector) Register() {
+	prometheus.MustRegister(c.metric)
+}
+
+func (c *DurationHistogramCollector) Before(ctx context.Context, method string, start time.Time) {
+}
+
+func (c *DurationHistogramCollector) After(ctx context.Context, method, statusCode string, start time.Time) {
+	durationBucket, ok := ctx.Value(queryRangeBucket).(string)
+
+	if !ok {
+		durationBucket = "null"
```

Comment:
a `null` bucket seems doesn't make sense, let's make it `Invalid` and add an error log.